ast.go

```diff
@@ -342,8 +342,29 @@ func newAssignStmtStructsAndPointers(
 	case !leftPtr && rightPtr:
 		// Pointer to Value
 		//
-		// <left> = *<right>
-		return astAssign(left, &ast.StarExpr{X: right})
+		// if <right> != nil {
+		//   <left> = *<right>
+		// } else {
+		//   <left> = nil
+		// }
+
+		leftPtrType, leftPtr := leftType.(*ast.StarExpr)
+
+		leftRealType := leftType
+		if leftPtr {
+			leftRealType = leftPtrType.X
+		}
+
+		return &ast.IfStmt{
```

Comment:
This lifted code from the existing area of code that handled the struct-to-struct conversions, which are different than basic type conversions for Go reasons.